cli/cmd/project/delete.go

```diff
@@ -22,17 +25,44 @@ func DeleteCmd(cfg *config.Config) *cobra.Command {
 			}
 			defer client.Close()
 
+			if !cmd.Flags().Changed("project") {
+				name, err = inferProjectName(cmd.Context(), client, cfg.Org, path)
+				if err != nil {
+					return err
+				}
+			}
+
+			if !force {
+				fmt.Printf("Warn: Deleting the project %q will remove all metadata associated with the project\n", name)
+
+				msg := fmt.Sprintf("Enter %q to confirm deletion", name)
+				project, err := cmdutil.InputPrompt(msg, "")
+				if err != nil {
+					return err
+				}
+
+				if project != name {
+					return fmt.Errorf("Entered incorrect name : %s", name)
+				}
+			}
+
 			_, err = client.DeleteProject(context.Background(), &adminv1.DeleteProjectRequest{
 				OrganizationName: cfg.Org,
-				Name:             args[0],
+				Name:             name,
 			})
 			if err != nil {
 				return err
 			}
 
-			cmdutil.SuccessPrinter(fmt.Sprintf("Deleted project: %v\n", args[0]))
+			cmdutil.SuccessPrinter(fmt.Sprintf("Deleted project: %v\n", name))
 			return nil
 		},
 	}
+
+	deleteCmd.Flags().SortFlags = false
+	deleteCmd.Flags().BoolVar(&force, "force", false, "Delete forcefully, skips the confirmation")
+	deleteCmd.Flags().StringVar(&name, "project", "", "Name")
+	deleteCmd.Flags().StringVar(&path, "path", ".", "Project directory")
```

Comment:
The default `org` should be `cfg.Org` but there should be a flag to override that. Applied to other project sub commands as well.